### DISCUSSION TITLE
Set session cookie for requests

### DISCUSSION THREAD

**Discussion by jinsley8** [hoppscotch/hoppscotch#3738] [Q&A]
I am trying to figure out how to best set the session cookie for requests in Hoppscotch.

I am using `next-auth` and on localhost the session cookie name is `next-auth.session-token` while the value is `eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0...`

For the requests I am setting a `Cookie` header then the value as `next-auth.session-token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0...`.

This doesn't seem to be working. Any tips on how to set the session cookie to send with the requests?

UPDATE:
It works from the browser version of Hoppscotch using the browser extension. I can't get it to work from the MacOS Desktop app.

**Comment by jinsley8** [comment]
Does anyone know how to correct use cookies in requests with the desktop app? Step by step?

The [doc says](https://docs.hoppscotch.io/documentation/features/cookies) cookies support is only available in the desktop app but I can still only get it to work in the browser app with the extension.

1. I follow the [steps here](https://docs.hoppscotch.io/documentation/features/cookies) set the localhost domain (`http://localhost:2000`) in the cookies

2. Then add cookie string `next-auth.session-token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0...;`.

3. Now when I make the request it still doesn't set the cookie

**Reply by jamesgeorge007** [reply]
Hi @jinsley8, can you try adding `localhost` as the domain name without the port number/protocol and giving it a shot again?

**Comment by vos-orb** [comment]
august2025 still not setting session cookie to desktop App (win)

**Comment by fenilli** [comment]
Did people figure it out? for now I'm doing it manually by getting the response headers and copy pasting the cookie.

It should do this automatically for all routes that are run to be honest.

**Reply by danielehrhardt** [reply]
push
